roles/servicemesh/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+      kind: ServiceMeshControlPlane
+      name: '{{ meta.name }}'
+      namespace: "{{ controlplane_namespace }}"
+    register: smcp_state
+
+  - set_fact:
+      smcp_ready: "{{ smcp_state | json_query('resources[0].status.conditions[]') | selectattr('type','match','Ready') | selectattr('status', 'match', 'True') | list | length > 0 }}"
+
+  - name: Progress to deploy
+    when: hf_state.resources[0].status.status == "Ready" and smcp_ready
+    operator_sdk.util.k8s_status:
+      api_version: ripsaw.cloudbulldozer.io/v1alpha1
+      kind: Benchmark
+      name: "{{ meta.name }}"
+      namespace: "{{ operator_namespace }}"
+      status:
+        state: "Deploy"
+        complete: false
+
+- name: Update current state
+  k8s_info:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: '{{ meta.name }}'
+    namespace: '{{ operator_namespace }}'
+  register: resource_state
+
+- name: Check ingress route
+  k8s_info:
+    api_version: route.openshift.io/v1
+    kind: Route
+    name: istio-ingressgateway
+    namespace: '{{ controlplane_namespace }}'
+  register: ingress_route
+
+- name: Set app route
+  when: ingress_route.resources | length > 0
+  set_fact:
+    #TODO: maybe use trunc_uuid in the hostname?
+    app_route: "{{ meta.name }}-app-{{ trunc_uuid}}{{ ingress_route.resources[0].status.ingress[0].host | regex_replace('^[^.]*', '')}}"
+
+- when: resource_state.resources[0].status.state == "Deploy"
+  block:
+  - name: Generate certificates
+    command: openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/app.key -out /tmp/app.crt -subj "/CN={{ app_route }}/O=app"
+
+  - name: Create ingressgateway certs secret
+    k8s:
+      definition:
+        apiVersion: v1
+        kind: Secret
+        metadata:
+          name: istio-ingressgateway-certs
+          namespace: "{{ controlplane_namespace }}"
+          ownerReferences: "{{ owner }}"
+        data:
+          tls.key: "{{ lookup('file', '/tmp/app.key') | b64encode }}"
+          tls.crt: "{{ lookup('file', '/tmp/app.crt') | b64encode }}"
+
+  - name: Get ingress gateway pods
+    k8s_info:
+      api_version: v1
+      kind: Pod
+      namespace: "{{ controlplane_namespace }}"
+      label_selectors:
+      - istio = ingressgateway
+    register: ingressgateways
+
+  - name: Reboot ingress gateway(s)
+    with_items: "{{ ingressgateways.resources | map(attribute='metadata.name') | list }}"
+    k8s:
+      state: absent
+      api_version: v1
+      kind: Pod
+      name: "{{ item }}"
+      namespace: "{{ controlplane_namespace }}"
+
+  - name: Create deployments
+    k8s:
+      definition: "{{ lookup('template', 'dc.yaml.j2') | from_yaml }}"
+
+  - name: Create routing resources
+    k8s:
+      definition: "{{ lookup('template', 'routing.yaml.j2') | from_yaml }}"
+
+  - name: Fetch pods
+    k8s_info:
+      api_version: v1
+      kind: Pod
+      namespace: '{{ workload_namespace }}'
+      label_selectors:
+      - app = app
```

Comment:
Small nit: k8s_info module accepts a field_selectors: argument. You can add
```
    field_selectors:
      - status.containerStatuses.ready=true
```

To simply the set_fact task below:
```
 ready_containers: "{{ pods.resources|length }}"
```

Reply:
@rsevilla87 Thanks, thought IIUC that wouldn't work 100% - such field selector would probably select pods that have at least one ready container, while I need both containers to be ready. Or do I misunderstand?

Reply:
Absolutely, I wasn't aware there are pods having several containers here.